spider/tuoitre.py

```diff
@@ -0,0 +1,48 @@
+from functools import partial
+from os.path import basename, splitext
+from urllib.parse import urldefrag, urlparse
+
+from html5lib import parse
+from httpx import ConnectTimeout
+from trio import open_file
+
+INDEX = 'https://tuoitre.vn/vaccine.html'
+
+parse_html5 = partial(parse, namespaceHTMLElements=False)
+
+
+def articles(links):
+    for a in links:
+        url, fragment = urldefrag(a.get('href'))
```

Comment:
Since the braindead at Tuổi Trẻ think it's OK to have `<a>` w/o `href`, I guess we'll have to work around them

```suggestion
        url = a.get('href')
        if url is None: continue
```

Defragmentation is not needed in this case (I don't see any `htm#` in the source).